nupic/encoders/scalar.py

```diff
@@ -156,561 +62,17 @@ def __init__(self,
-    encoding = self._topDownMappingM.getRow(category)
-
-    # Which input value does this correspond to?
-    if self.periodic:
-      inputVal = (self.minval + (self.resolution / 2.0) +
-                  (category * self.resolution))
-    else:
-      inputVal = self.minval + (category * self.resolution)
-
-    return [EncoderResult(value=inputVal, scalar=inputVal, encoding=encoding)]
-
-
-
-  ############################################################################
-  def topDownCompute(self, encoded):
-    """ See the function description in base.py
-    """
-
-    # Get/generate the topDown mapping table
-    topDownMappingM = self._getTopDownMapping()
-
-    # See which "category" we match the closest.
-    category = topDownMappingM.rightVecProd(encoded).argmax()
-
-    # Return that bucket info
-    return self.getBucketInfo([category])
-
-
-  ############################################################################
-  def closenessScores(self, expValues, actValues, fractional=True):
-    """ See the function description in base.py
-    """
-
-    expValue = expValues[0]
-    actValue = actValues[0]
-    if self.periodic:
-      expValue = expValue % self.maxval
-      actValue = actValue % self.maxval
-
-    err = abs(expValue - actValue)
-    if self.periodic:
-      err = min(err, self.maxval - err)
-    if fractional:
-      pctErr = float(err) / (self.maxval - self.minval)
-      pctErr = min(1.0, pctErr)
-      closeness = 1.0 - pctErr
-    else:
-      closeness = err
-
-    return numpy.array([closeness])
-
-
-  ############################################################################
-  def dump(self):
-    print "ScalarEncoder:"
-    print "  min: %f" % self.minval
-    print "  max: %f" % self.maxval
-    print "  w:   %d" % self.w
-    print "  n:   %d" % self.n
-    print "  resolution: %f" % self.resolution
-    print "  radius:     %f" % self.radius
-    print "  periodic: %s" % self.periodic
-    print "  nInternal: %d" % self.nInternal
-    print "  rangeInternal: %f" % self.rangeInternal
-    print "  padding: %d" % self.padding
+    """backwards compatibility constructor - from SimpleScalarEncoder"""
+ 
+    "get resolution by constructing a dummy SimpleScalar encoder and asking its resolution"
+    dummySimpleScalar = SimpleScalarEncoder(w, minval, maxval, periodic=periodic,
+                                            n=n, radius=radius, resolution=resolution,
+                                            name=name, verbosity=verbosity,
+                                            clipInput=clipInput, forced=forced)
+    res = dummySimpleScalar.resolution
+    super(ScalarEncoder, self).__init__(res,
+                                   #     w=w, #TODO should we: a) add 'forced' to RDSE and allow given w,n; or
+                                              #                b) pass only resolution and not w,n from former SimpleScalar to RDSE ?
+                                   #     n=n,
```

Comment:
@chetan51 before any work on the tests, we need to decide how to handle the wrapper constructor from old ScalarEncoder: 
`RDSE(resolution, n=opt, w=opt)` vs `Scalar(w, minval, maxval, n=opt, resolution=opt)` - what to do with the optional params to RDSE? 

* We could ignore them completely (and put warning in the init() so users dont swarm over them!:) )
* use just `w` and set `n` to n=11*w which is said to be optimal value for RDSE.